src/v/storage/segment_appender.cc

```diff
@@ -50,11 +51,14 @@ size_missmatch_error(const char* ctx, size_t expected, size_t got) {
       "{}. Size missmatch. Expected:{}, Got:{}", ctx, expected, got));
 }
 
+static constexpr auto head_sem_name = "s-appender-head";
+
 segment_appender::segment_appender(ss::file f, options opts)
   : _out(std::move(f))
   , _opts(opts)
   , _concurrent_flushes(ss::semaphore::max_counter())
-  , _prev_head_write(ss::make_lw_shared<ss::semaphore>(1))
+  , _prev_head_write(
+      ss::make_lw_shared<>(ssx::make_named_semaphore(1, head_sem_name)))
```

Comment:
`ssx::(named)_mutex`?

Reply:
Are we comfortable with this `mutex` naming in general? IIUC a `semaphore(1)` is equivalent to an unlocked mutex, except it breaks if there are multiple unlocks (i.e. value can go above 1).

Reply:
I didn't check if they should be a mutex, I just posed the question, since it started at 1. Feel free to ignore all the mutex comments, that could easily be left to another PR, even if they should be a mutex.